python/llm/src/bigdl/llm/transformers/models/utils.py

```diff
@@ -295,3 +295,19 @@ def use_fused_layer_norm(x: torch.Tensor, training: bool):
             or x.reshape(-1, x.size(-1)).size(0) == 1
         )
     )
+
+
+def fp16_fusion_check(proj, x, training):
+    # only use fp16 fusion on PVC inference
+    if proj.qtype != ggml_tensor_qtype["fp16"]:
+        return False
+    if proj.weight_type != 2:
+        return False
+    if training:
+        return False
+    if x.requires_grad:
+        return False
+    device_type = get_xpu_device_type(x)
```

Comment:
may check device type is 'xpu' first, if calling this function with a 'cpu' tensor, it will throw an error. or change this function and make it return 'cpu' when the tensor is not on 'xpu'

Reply:
fix it